src/ucs/sys/sock.c

```diff
@@ -391,6 +392,25 @@ ucs_status_t ucs_sockaddr_set_port(struct sockaddr *addr, uint16_t port)
     }
 }
 
+int ucs_sockaddr_is_inaddr_any(struct sockaddr *addr)
+{
+    struct sockaddr_in6 *addr_in6;
+    struct sockaddr_in *addr_in;
+
+    switch (addr->sa_family) {
+    case AF_INET:
+        addr_in = (struct sockaddr_in *)addr;
+        return addr_in->sin_addr.s_addr == INADDR_ANY;
+    case AF_INET6:
+        addr_in6 = (struct sockaddr_in6 *)addr;
+        return !memcmp(&addr_in6->sin6_addr, &in6addr_any, sizeof(addr_in6->sin6_addr));
```

Comment:
can use `UCS_SOCKET_INET6_ADDR(addr)`/`UCS_SOCKET_INET_ADDR(addr)` instead
so all `case` will be one-liners

Reply:
@dmitrygx Thanks for the quick review. Can you review my changes?